test/resources/national_collection.yml

```diff
@@ -28,7 +28,11 @@
   - column: SQU03_6_2_2:N
     mappings:
     - field: SQU03_6_2_2
-      arrow_data_type: :int32
+      arrow_data_type: :list
```

Comment:
I still do wonder if a format identifier like `list[int]` or `[:int]` might not be a bit more idiomatic here, although granted would probably cause other internal issues with the symbol representation. The latter would still be valid YAML I think, representing a literal array.

Reply:
@joshpencheon would you have something like:

```
arrow_data_type: [:string]
arrow_list_field:
  name: code_name
  split: ';'
```

Reply:
That's a good point; I think I'd been thinking along the lines of this being an expression of datatype, whereas actually it's more like a processing directive so perhaps discount my earlier feedback!